torch_xla/csrc/helpers.cpp

```diff
@@ -691,6 +755,109 @@ xla::XlaOp XlaHelpers::ImplicitBroadcast(xla::XlaOp op,
   return new_op;
 }
 
+std::pair<xla::XlaOp, xla::XlaOp>
+XlaHelpers::ImplicitBroadcastWithUnboundedDynamicShapes(
+    xla::XlaOp op1, xla::XlaOp op2, const xla::Shape& shape) {
+  XLA_CHECK(shape.is_unbounded_dynamic());
+
+  const xla::Shape& shape1 = ShapeHelper::ShapeOfXlaOp(op1);
+  const xla::Shape& shape2 = ShapeHelper::ShapeOfXlaOp(op2);
+  const auto& shape1_dims = shape1.dimensions();
+  const auto& shape2_dims = shape2.dimensions();
+  const auto& shape_dims = shape.dimensions();
+
+  XLA_CHECK(shape_dims.size() ==
+            std::max(shape1_dims.size(), shape2_dims.size()));
+
+  int64_t size_delta = shape2_dims.size() - shape1_dims.size();
```

Comment:
How about `rank_delta`, since size could also refer to the size of a dimension

Reply:
will update the code. We may not need that.